app/views/leagues/show.html.haml

```diff
@@ -21,19 +21,23 @@
             class: 'btn btn-default pull-right'
 
   - matches = @personal_matches || @top_div_matches
-  - if !matches.empty?
-    .panel-heading
-      - if @personal_matches
-        Your Upcoming Matches
+
+  .panel-heading
+    - if user_signed_in? && @roster
+      - if !@personal_matches.empty?
+        Upcoming matches for your team:
+
       - else
-        Upcoming Matches for #{@divisions.first.name}
+        There are no upcoming matches for your team, #{@roster.name}.
```

Comment:
I think making the roster name a link to the team would be better. Something like:
```
There are no upcoming matches for your team: #{present(@roster).link}.
```